webpack/assets/javascripts/react_app/components/HostStatuses/HostStatusesSelectors.js

```diff
@@ -0,0 +1,72 @@
+import { selectAPIResponse } from '../../redux/API/APISelectors';
+import {
+  GLOBAL_STATUS_OK,
+  GLOBAL_STATUS_WARN,
+  GLOBAL_STATUS_ERROR,
+} from './constants';
+
+export const selectManipulatedStatuses = (state, key) => {
```

Comment:
Selectors work the best as a moudlar structure, so we can reuse them in other places, plus it easier to read, maintain (optimize) and test: 
```js
// can be a predefined key
import { HOST_STATUS_KEY } from './consts'; 

export const selectHostStatuses = state => {
  const response = selectAPIResponse(state, HOST_STATUS_KEY);
}

export const selectHostStatusesDetails = state => {
   selectHostStatuses(state)?.results?.details
}

export const selectGlobalStatus = state =>  {
  const withTotal = 
    selectHostStatusesDetails(state)
      .filter(({ total }) => total > 0)
      .map(({ global_status: gs }) => gs),
    return Math.max(withTotal, 0);
}

export const selectOKStatueses = state => {
// ...
} 

// etc
```